src/pip/_internal/utils/models.py

```diff
@@ -5,11 +5,22 @@
 
 import operator
 
+from pip._internal.utils.typing import MYPY_CHECK_RUNNING
 
-class KeyBasedCompareMixin(object):
+if MYPY_CHECK_RUNNING:
+    from typing import List
+
+
+class Base(object):
+    __slots__ = []  # type: List[str]
```

Comment:
Why is this base class needed? A quick test shows me that Mypy is able to pick up the correct type by inheriting `object`:

```
$ cat x.py
from typing import List

class Child(object):
    __slots__ = [123]
$ mypy x.py
x.py:4: error: List item 0 has incompatible type "int"; expected "str"
```